src/Stateless/DotGraph/DotGraphFormatter.cs

```diff
@@ -7,78 +7,237 @@ namespace Stateless.DotGraph
+
         /// <summary>
-        /// Produces a DOT GraphViz graph.
+        /// A string representation of the stateRepresentation machine in the DOT graph language.
         /// </summary>
-        /// <param name="stateMachineInfo">The StateMachineInfo to be mapped.</param>
-        /// <returns>DOT GraphViz text.</returns>
-        public static string Format(StateMachineInfo stateMachineInfo)
+        /// <returns>A description of all simple source states, triggers and destination states.</returns>
+        public string ToDotGraph(StateMachineInfo machineInfo)
         {
-            var bindings = stateMachineInfo.States;
+            clusterDictionary = new Dictionary<string, string>();
+            _handledStates = new List<string>();
+            _formattedNodesList = new List<string>();
+
+            FillClusterNames(machineInfo);
 
+            string dirgraphText = $"digraph {{{System.Environment.NewLine}compound=true;{System.Environment.NewLine}rankdir=\"LR\"{System.Environment.NewLine}";
+
+            // Start with the clusters
+            foreach (StateInfo stateInfo in machineInfo.States.Where(v => v.Substates.Count() > 0))
+            {
+                dirgraphText += BuildNodesRepresentation(stateInfo);
+            }
+
+            // Next process all non-cluster elements
+            foreach (var stateRep in machineInfo.States.Where(v => v.Superstate == null && v.Substates.Count() == 0))
+            {
+                dirgraphText += BuildNodesRepresentation(stateRep);
+            }
+            // now build behaviours
+            foreach (StateInfo stateInfo in machineInfo.States)
+            {
+                var behaviours = ProcessTriggerBehaviour(stateInfo).ToArray();
+                if (behaviours.Length > 0)
+                    dirgraphText += $"{System.Environment.NewLine}{string.Join(System.Environment.NewLine, behaviours)} ";
+            }
+
+            dirgraphText += $"{System.Environment.NewLine}}}";
+            return dirgraphText;
+
+        }
+
+
+        private List<string> ProcessEntries(StateInfo stateInfo)
+        {
             List<string> lines = new List<string>();
-            List<string> unknownDestinations = new List<string>();
+            foreach (InvocationInfo entryAction in stateInfo.EntryActions)
+                lines.Add(entryAction.Description);
+            return lines;
+        }
+        private List<string> ProcessExits(StateInfo stateInfo)
+        {
+            List<string> lines = new List<string>();
+            foreach (InvocationInfo exitAction in stateInfo.ExitActions)
+                lines.Add(exitAction.Description);
+            return lines;
+        }
+
+        private void FillClusterNames(StateMachineInfo machineInfo)
+        {
+            int i = 0;
 
-            foreach (var binding in bindings)
+            foreach (StateInfo stateInfo in machineInfo.States.Where(sc => sc.Substates != null && sc.Substates.Count() > 0))
```

Comment:
Nit: `var`.

Also: `Where(sc => sc?.Substates.Count() > 0)` should work here (unsure about `Any()` without more syntactic noise).